File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int n= prices.length;
          if(n==1){
            return 0;
          }
        int ans=0;
         int min_so_far=prices[0];
        for(int i=1; i<n; i++){
           min_so_far = Math.min(min_so_far, prices[i]);
            ans= Math.max(ans, prices[i]- min_so_far);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n= s.size();

        unordered_map<char, int>mp;
           int ans=0;
        int i=0;
        for(int j=0; j<n; j++){
            mp[s[j]]++;
          
            if(mp[s[j]]>1){
               while(mp[s[j]]>1){
                    mp[s[i]]--;
                    i++;
               }
            }
               ans= max(ans, j-i+1);
        }

          return ans;
    }
};
